websites/M/MarineTraffic/presence.ts

```diff
@@ -0,0 +1,330 @@
+		"users/plans": { state: "Online Services & Plans" },
+		"maritime-companies/index": { state: "Business Profile" },
+		"corporate/index": { state: "Corporate Account" },
+		"users/settings": { state: "User Settings" },
+		"notifications/index": { state: "User Notifications" },
+	};
+
+	for (const [path, data] of Object.entries(pages)) {
+		if (document.location.href.includes(path))
+			presenceData = { ...presenceData, ...data };
+	}
+
+	if (document.location.hostname.startsWith("shop")) {
+		presenceData.details = "Browsing AIS e-shop:";
+		presenceData.state = document.title.split("-")[0];
+	} else if (document.location.hostname.startsWith("help")) {
+		presenceData.details = "Reading help:";
+		presenceData.state = document.title.split("–")[0];
+		presenceData.buttons = [
+			{ label: "Read this page", url: document.location.href },
+		];
+	} else if (document.location.pathname.startsWith("/blog")) {
+		presenceData.details = "Reading a blog post:";
+		presenceData.state = document.title.split("-")[0];
+		presenceData.buttons = [
+			{ label: "Read this post", url: document.location.href },
+		];
+	} else if (document.location.pathname.includes("ais/home/center")) {
+		presenceData.details = "Viewing live map";
+		setInterval(() => {
+			locationMap = document
+				.querySelector(
+					"#map_canvas > div.leaflet-control-container > div.leaflet-top.leaflet-right.leaflet-mouse-position-control > div.leaflet-control-mouseposition.leaflet-control"
+				)
+				.innerHTML.split("<br>")[0];
+		}, 5000);
+		presenceData.state = `Near ${locationMap}`;
+		presenceData.buttons = [
+			{ label: "View Page", url: document.location.href },
+		];
+	} else if (document.location.pathname.includes("ais/home/shipid")) {
+		elemName = capitalize(
+			document
+				.querySelector(
+					"#app > div > div.jss2 > div.d-flex.align-items-start.justify-content-start.flex-1-1-auto.home-layout > div.d-flex.flexcol.align-items-start.justify-content-start.flex-wrap.flex-1-1-auto > div > div:nth-child(4) > div > div > div > div > div > div.handle.cursor-move > div > div > div.d-flex.flexcol.align-items-start.justify-content-start.flex-wrap.flex-1-1-auto > div > a"
+				)
+				.innerHTML.toLowerCase()
+		);
+		shipStatus = document.querySelector(
+			"#infosnippet_container > div > div > div > div:nth-child(2) > div > div:nth-child(1) > div > strong"
+		).innerHTML;
+
+		if (
+			document
+				.querySelector<HTMLAnchorElement>(
+					"#infosnippet_container > div > div > div > div:nth-child(2) > div > div > div.d-flex.align-items-center.justify-content-start.flex-wrap.flex-1-1-auto > div:nth-child(1) > div > span > a"
+				)
+				.href.includes("javascript")
+		)
+			shipDep = "Unknown";
+		else {
+			shipDep = capitalize(
+				document
+					.querySelector<HTMLAnchorElement>(
+						"#infosnippet_container > div > div > div > div:nth-child(2) > div > div > div.d-flex.align-items-center.justify-content-start.flex-wrap.flex-1-1-auto > div:nth-child(1) > div > span > a"
```

Comment:
![image](https://user-images.githubusercontent.com/42322979/187645296-40a53d65-546a-46ff-b330-2f57efed7056.png)
```suggestion
						'[class="departure-arrival-label"]'
```